### PR TITLE
Reduce Codecov noise

### PR DESCRIPTION
**Pull request by alecandido**

### CODE REVIEW

**Comment by alecandido** [comment]
@renatomello this should prevent noise from your single line issue (it doesn't solve the uncovered line, but it will avoid bothering you)

**Review by andrea-pasquale — APPROVED** [review]
Thanks @alecandido.

File: .github/codecov.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    project:
+      default:
+        threshold: 0.5%
+
+github_checks:
+  annotations: false
+
+ignore:
+  - "setup.py"
+  - "src/qibo/tests/test_custom_functions.py"
```
Comment (stavros11) [inline .github/codecov.yml:12]: I know it existed before this PR, but I have the impression that these files do not exist anymore, so we don't need these lines.

**Review by stavros11 — COMMENTED** [review]
What is the motivation of not failing when a single line is not covered? I agree that it is annoying but may be the only way to maintain coverage to 100% (excluding `# pragma: no cover` ofcourse).

File: .github/codecov.yml (hunk 1 above)
Comment (alecandido) [reply .github/codecov.yml]: You're right, I simply ignored the issue. But it costs little, so it's worth to do it.

Solved in 47b2518

**Comment by alecandido** [comment]
> What is the motivation of not failing when a single line is not covered? I agree that it is annoying but may be the only way to maintain coverage to 100% (excluding `# pragma: no cover` ofcourse).

To be fair, the main motivation is that I do not care too much about the 100% coverage.

Coverage is definitely useful, but `99% == 100%` in all respects.
(you could always cover, but not test effectively, or not test it, but with little to no impact)

To me, it's more relevant that sensible PRs are marked with the green tick, while the red cross is reserved to those failing tests, or with a sizeable drop in coverage. We could still target the 100%, but I would not do it PR by PR (e.g. if for a release we don't have 100%, and we'd want it, we could make an effort to compensate).

However, the 100% could also be achieved through human revision, paying more attention to tests and suggesting meaningful ones.

**Review by stavros11 — APPROVED** [review]
> To be fair, the main motivation is that I do not care too much about the 100% coverage.

Thanks, that was what I was expecting, just wanted to make sure.

> Coverage is definitely useful, but `99% == 100%` in all respects. (you could always cover, but not test effectively, or not test it, but with little to no impact)

I agree about the effectiveness of tests. The only advantage of covering 100% is that you are at least sure that every line can be executed, so there are no obvious issues (many of which would be captured by pylint anyway). For sure it's far from a guarantee that the code does what is supposed to.

Anyway, I agree with merging this PR.

**Pull request merged**